lib/decompress/zstd_decompress.c

```diff
@@ -2059,7 +2059,7 @@ size_t ZSTD_decompressStream(ZSTD_DStream* zds, ZSTD_outBuffer* output, ZSTD_inB
                     if (ZSTD_isError(decompressedSize)) return decompressedSize;
                     DEBUGLOG(4, "shortcut to single-pass ZSTD_decompress_usingDDict()")
                     ip = istart + cSize;
-                    op += decompressedSize;
+                    op = op ? op + decompressedSize : op;
```

Comment:
Detailing the thought process :

That such a scenario can even be triggered in `ubsan` tests feels strange initially: 
for execution to reach this code position, it needs to pass the test : 
`(oend-op) >= zds->fParams.frameContentSize`.
So, if there is any content, then it's required that `op != NULL`.

However, thinking about edge cases,
it's indeed possible that `frameContentSize==0`.
This is an empty frame (yet still a valid frame).
It's obviously not an expected outcome in general, the point is that it _can_ happen, it's allowed.
In which case, it becomes technically possible to "decompress" such an empty frame into `NULL`.

Now, there are more philosophical questions associated to this scenario : 
Is it normal to ask to decompress a frame into a `NULL` buffer ?
Should it be an instant fail ? or should it succeed in the specific case where `frameContentSize==0` ?

Note that this code block is just an "optimization" : it triggers single-pass mode when space conditions are met, which requires less workload than full-featured streaming mode.
Which means that skipping this block is not an "end" sentence : it just requires to employ the more complex streaming mode.

Therefore, one possible idea here could be to add a `op != NULL` condition to enter this code block,
so that `op==NULL` is no longer a possible scenario to support inside the code block
(it must employ the more general streaming code path).
However, such a solution does not appear "obviously" better than the ternary operator you selected.

I would say, the only piece of information missing is why the scenario (`op==NULL`) can technically happen here, justifying the existence of this ternary operator to any future contributor reviewing the code.